### PR TITLE
Eis api endpoints

### PR DESCRIPTION
**Pull request by mattrobertsky**

### CODE REVIEW

File: app/uk/gov/hmrc/eusubsidycompliancestub/controllers/UndertakingController.scala [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+              "retrieveUndertakingResponse" -> Json.obj(
+                "responseCommon" ->
+                  ResponseCommon(
+                    EisStatus.NOT_OK,
+                    EisStatusString("String"), // taken verbatim from spec
+                    LocalDateTime.now,
+                    List(
+                      Params(
+                        EisParamName.ERRORCODE,
+                        EisParamValue("107")
```
Comment (adam-dye) [inline app/uk/gov/hmrc/eusubsidycompliancestub/controllers/UndertakingController.scala:94]: Don't we need the other error codes too. 
003 Request could not be processed
055 ID number missing or invalid
067 ID type missing or invalid

File: app/uk/gov/hmrc/eusubsidycompliancestub/models/Address.scala [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+// * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// * See the License for the specific language governing permissions and
+// * limitations under the License.
+// */
+//
+//package uk.gov.hmrc.eusubsidycompliancestub.models
+//
+//import uk.gov.hmrc.eusubsidycompliancestub.models.types.{AddressLine, CountryCode, Postcode}
+//
+//case class Address(
```
Comment (adam-dye) [inline app/uk/gov/hmrc/eusubsidycompliancestub/models/Address.scala:37]: Delete this file if we're sure we're not using it

File: app/uk/gov/hmrc/eusubsidycompliancestub/models/BusinessEntity.scala [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+package uk.gov.hmrc.eusubsidycompliancestub.models
+
+import play.api.libs.json.{Json, OFormat}
+import uk.gov.hmrc.eusubsidycompliancestub.models.types.EORI
+
+case class BusinessEntity(
+  businessEntityIdentifier: EORI,
+  leadEORI: Boolean,
+//  address: Option[Address], // not needed but still included in EIS API...
```
Comment (adam-dye) [inline app/uk/gov/hmrc/eusubsidycompliancestub/models/BusinessEntity.scala:25]: And remove this if we're not using it

File: project/AppDependencies.scala [hunk 4]
```diff
@@ -1,17 +1,34 @@
...
   val compile = Seq(
-    "uk.gov.hmrc"             %% "bootstrap-backend-play-28"  % "5.13.0"
+    "uk.gov.hmrc"             %% "bootstrap-backend-play-28"  % "5.13.0",
+    "org.typelevel"           %% "cats-core"                  % "2.6.1",
+    "com.chuusai"             %% "shapeless"                  % "2.3.7",
+    "uk.gov.hmrc"             %% "stub-data-generator"        % "0.5.3",
+    "com.github.fge"          %  "json-schema-validator"      % "2.2.6"
   )
 
   val test = Seq(
```
Comment (adam-dye) [inline project/AppDependencies.scala:16]: Sort out commented libs

File: test/uk/gov/hmrc/eusubsidycompliancestub/controllers/BaseControllerSpec.scala [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+import org.scalatestplus.play.PlaySpec
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+import play.api.http.HeaderNames
+import play.api.mvc.Results
+import play.api.test.FakeHeaders
+
+class BaseControllerSpec extends
+  PlaySpec
+  with MockitoSugar // TODO remove if unneeded
```
Comment (adam-dye) [inline test/uk/gov/hmrc/eusubsidycompliancestub/controllers/BaseControllerSpec.scala:29]: Is this needed?

**Review by adam-dye — CHANGES_REQUESTED** [review]

**Review by adam-dye — APPROVED** [review]

**Review by hmrclt — APPROVED** [review]

**Pull request merged**